hangman_game.py

```diff
@@ -0,0 +1,128 @@
+                print('  |      ')
+            if 0 <= self.hangmanState <= 1:
+                print('  |   /|\ ')
+                if self.hangmanState == 1:
+                    print('  |   / ')
+                if self.hangmanState <= 0:
+                    print('  |   / \ ')
+            if self.hangmanState == 5:
+                print('  |    ')
+                print('  |    ')
+        else:
+            for i in range(3):
+                print('  |      ')
+        print('__|__\n')
+
+
+    def showSplit(self, array):
+        print(' '.join(array))  # format output for hangman
+
+    def playGame(self):
+        target = self.setupGame()  # specify target word with setupGame
+        targetSplit = np.array(list(target))  # created for finding correct guesses
+        maskedSplit = np.array(['_' for letter in list(target)])  # created to show state of hangman game
+
+        while self.hangmanState > 0:
+            self.showBoard()  # start off by showing state of hangman
+            self.showSplit(maskedSplit)  # display state of board
+            # print(target)  # used for debugging
+            letterGuess = input('Guess a letter ')
+            if letterGuess.upper() in self.validLetters:
+                if letterGuess.upper() in self.guessedLetters:
+                    self.hangmanState -= 1  # decrement for repeated guess
+                    print('Already guessed this letter')
+                else:
+                    if letterGuess.upper() in targetSplit:
+                        validSpots = np.where(targetSplit == letterGuess.upper())  # all indices of match
+                        self.guessedLetters.append(letterGuess.upper())  # add guess to list of guesses
+                        for index in validSpots:
+                            maskedSplit[index] = letterGuess.upper()  # replace masked spot with letter
+                        if '_' not in maskedSplit:
+                            print('You won the game!')
+                            self.showSplit(maskedSplit)  # show the completed word
+                            self.showBoard()
+                            self.hangmanState = -1  # exit the game since word was correctly guessed
+                    else:
+                        self.hangmanState -= 1  # decrement hangman state for incorrect guess
+                        print('Incorrect guess')
+                        self.guessedLetters.append(letterGuess.upper())
+            else:
+                print('Invalid input, guess a letter')
+        else:
+            print('Game Over!')
+            self.showSplit(maskedSplit)  # show the masked split
+            self.showBoard()  #  show the board
+            print('Correct word was {}'.format(target))  # tell user what correct word was
+
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Input a minimum word length for hangman')  # create object
+    parser.add_argument("-l", '--minLength', type=int, required=True,
```

Comment:
may want to consider using a different letter than lowercase L since it looks like a 1 in the terminal